frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp

```diff
@@ -265,9 +273,85 @@ MlirType AcapController::mapIValueToMlirType(MlirLocation loc,
   if (ival.isTensor()) {
     return typeMapper.forwardTensorToType(ival.toTensor());
   }
+  if (ival.isBool()) {
+    // TODO: Switch to the numpy.bool type as that is a closer domain match.
+    return mlirIntegerTypeGet(funcBuilder->getContext(), 1);
+  }
   return {nullptr};
 }
 
+MlirValue AcapController::importTensorByValue(at::Tensor tensor) {
+  using at::ScalarType;
+
+  auto throwUnsupportedTensorError = [&]() {
+    std::stringstream msg;
+    msg << "Unsupported import tensor type: " << tensor;
+    throw std::invalid_argument(msg.str());
+  };
+
+  // Get a C-contiguous form as we can bulk-load that into a DenseElementsAttr.
+  if (!tensor.is_contiguous())
+    tensor = tensor.contiguous();
+
+  // The flat number of bytes throws an exception for tensors that are not
+  // dense and accessible as such.
+  (void)tensor.nbytes();
```

Comment:
They do have some facilities for this kind of assertions, e.g. at::check* functions:
https://github.com/llvm/mlir-npcomp/blob/b6bdc8cc4fcb555510901148afbc222735026662/tools/mnist-playground/mnist-playground.cpp#L79